benchmark/reformat_input.py

```diff
@@ -0,0 +1,407 @@
+
+def reformat_trex(gene_tree,
+                  species_tree,
+                  output_tree_fp):
+    """ Reformat input trees to the format accepted by T-REX
+
+    Binary trees only, leaves of species and gene trees must have equal names
+
+    Parameters
+    ----------
+    gene_tree: skbio.TreeNode
+        TreeNode instance for gene tree
+    species_tree_fp: skbio.TreeNode
+        TreeNode instance for species tree
+    output_tree_fp: string
+        file path to output trees (species followed by gene)
+
+    See Also
+    --------
+    skbio.TreeNode
+    """
+    # trim gene tree leaves to exclude '_GENENAME' (if exists)
+    trim_gene_tree_leaves(gene_tree)
+    # join species and gene tree into one file
+    join_trees(gene_tree,
+               species_tree,
+               output_tree_fp)
+
+
+def reformat_riatahgt(gene_tree,
+                      species_tree,
+                      output_tree_fp):
+    """ Reformat input trees to the format accepted by RIATA-HGT (PhyloNet)
+
+    Input to RIATA-HGT is a Nexus file. The number of leaves in the species
+    and gene tree must be equal with the same naming.
+
+    gene_tree: skbio.TreeNode
+        TreeNode instance for gene tree
+    species_tree_fp: skbio.TreeNode
+        TreeNode instance for species tree
+    output_tree_fp: string
+        file path to output trees (Nexus format)
+
+    See Also
+    --------
+    skbio.TreeNode
+    """
+    nexus_file = """#NEXUS
+BEGIN TREES;
+Tree speciesTree = SPECIES_TREE
+Tree geneTree = GENE_TREE
+END;
+BEGIN PHYLONET;
+RIATAHGT speciesTree {geneTree};
+END;
+"""
+    # trim gene tree leaves to exclude '_GENENAME' (if exists)
+    trim_gene_tree_leaves(gene_tree)
+    p = replace(nexus_file, 'SPECIES_TREE', str(species_tree)[:-1])
+    p = replace(p, 'GENE_TREE', str(gene_tree)[:-1])
+    with open(output_tree_fp, 'w') as output_tree_f:
+        output_tree_f.write(p)
+
+
+def reformat_jane4(gene_tree,
+                   species_tree,
+                   output_tree_fp):
+    """ Reformat input trees to the format accepted by Jane4
+
+    Input to Jane4 is a Nexus file, the trees cannot not contain
+    branch lengths and the species/gene leaves mapping is required
+
+    Parameters
+    ----------
+    gene_tree: skbio.TreeNode
+        TreeNode instance for gene tree
+    species_tree_fp: skbio.TreeNode
+        TreeNode instance for species tree
+    output_tree_fp: string
+        file path to output trees (Nexus format)
+
+    See Also
+    --------
+    skbio.TreeNode
+    """
+    nexus_file = """#NEXUS
+begin host;
+tree host = SPECIES_TREE
+endblock;
+begin parasite;
+tree parasite = GENE_TREE
+endblock;
+begin distribution;
+Range MAPPING;
+endblock;
+"""
+    # create a mapping between the species and gene tree leaves
+    mapping_dict = species_gene_mapping(gene_tree=gene_tree,
+                                        species_tree=species_tree)
+    # set branch lengths to None
+    for node in gene_tree.postorder():
```

Comment:
hmm I didn't want to because I may want to use ```species_gene_mapping()``` for another tool where I don't need to remove the branch lengths. However, I created a new function ```remove_branch_lengths()``` and moved the code there, which is also used by ranger-dtl.